providers/openebs/cspc_csi_migration/cstor_volume_migration.yml

```diff
@@ -0,0 +1,124 @@
+---
+            - name: Obtain the application namespace to scaledown the application
+              shell: >
+                 kubectl get pvc --all-namespaces --no-headers | grep {{ outer_item }} | awk '{print $1}'
+              args:
+                executable: /bin/bash
+              register: app_ns
+
+            - name: Obtain the application deployment name to scaledown the application
+              shell: >
+                 kubectl get deploy -n {{ app_ns.stdout }} --no-headers  -o custom-columns=:.metadata.name
+              args:
+                executable: /bin/bash
+              register: app_deploy_name
+
+            - name: Scaledown the application
+              shell: >
+                 kubectl scale deploy -n {{ app_ns.stdout }} {{ app_deploy_name.stdout }} --replicas=0
+              args:
+                executable: /bin/bash
+              register: scale_down_status
+              
+            - name: Verify if the application scaled down successfully
+              shell: >
+                 kubectl get deploy -n {{ app_ns.stdout }} --no-headers -o custom-columns=:.spec.replicas
+              args:
+                executable: /bin/bash
+              register: deploy_status
+              until: deploy_status.stdout|int == 0
+              delay: 2
+              retries: 30
+
+            - name: check if the application pod is deleted
+              shell: >
+                kubectl get pod -n {{ app_ns.stdout }} --no-headers -o custom-columns=:.status.phase
+              args:
+                executable: /bin/bash
+              register: status
+              until: "status.stdout == ''"
```

Comment:
this will fail for applications with multiple microservices. say Kafka. It will have Kafka and zookeeper. If you scaledown Kafka, zookeeper will still be present in the namespace